dali/pipeline/data/tensor_test.cc

```diff
@@ -425,6 +426,53 @@ TYPED_TEST(TensorTest, TestResize) {
   }
 }
 
+TYPED_TEST(TensorTest, TestCustomAlloc) {
+  Tensor<TypeParam> tensor;
+
+  // Get shape
+  auto shape = this->GetRandShape();
+  int allocations = 0;
+  std::function<void*(size_t)> alloc_func;
+  std::function<void(void *)> deleter;
+  if (std::is_same<TypeParam, CPUBackend>::value) {
+    alloc_func = [&](size_t bytes) {
+      return new uint8_t[bytes];
+    };
+    deleter = [&](void *ptr) {
+      free(ptr);
+      allocations--;
+    };
+  } else {
+    alloc_func = [](size_t bytes) {
+      void *ptr;
+      CUDA_CALL(cudaMalloc(&ptr, bytes));
+      return ptr;
+    };
+    deleter = [&](void *ptr) {
+      CUDA_DTOR_CALL(cudaFree(ptr));
+      allocations--;
+    };
+  }
+
+  tensor.set_alloc_func([&](size_t bytes) {
+    allocations++;
```

Comment:
why not doing this inside alloc_func?

Reply:
It doesn't matter much. It's just that I have to write this function anyway, so I can put have this line once. There's no such common place for deleters.